### PR TITLE
fix: change is owning reaction, separating anon and signed comment reaction

### PR DESCRIPTION
**Pull request by fajarism**
### Summary by CodeRabbit

---
- New Feature: Enhanced the chat functionality in feeds by introducing the ability to distinguish between anonymous and signed posts. This will allow users to better understand the context of the conversation.
- Refactor: Improved the logic for determining whether a post is owned by the current user, providing more accurate results and enhancing user experience.
- Bug Fix: Fixed an issue where the `getDetail` function was not receiving all necessary parameters, leading to incomplete or incorrect data being returned. This fix ensures that users always see the correct details associated with each post.
---

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 9638e663bf40e65529bb3f8373aa4dba98b5bf60 and 71b13d538d454255fc223ed8ccff1f15a427ed5d commits.
</details>

<details>
<summary>Files selected (2)</summary>

* controllers/feeds/getAnonymousFeedChat.js (2)
* controllers/feeds/getFeedChat.js (6)
</details>






---

File: controllers/feeds/getAnonymousFeedChat.js [hunk 1]
```diff
@@ -36,10 +36,11 @@ const getAnonymousFeedChatService = async (req, res) => {
         constantActor,
         isAnonym,
         isOwnSignedPost,
+        isOwnAnonymousPost,
         isOwnPost,
         message,
         actor
-      } = getDetail(req, b, mySignedId);
+      } = getDetail(req, b, mySignedId, req?.userId);
```
Comment (bastian-purrer) [inline controllers/feeds/getAnonymousFeedChat.js:43]: Image description   CodeRabbit

The `getDetail` function call has been updated with new arguments. However, it's not clear if the function definition has been updated to handle these new arguments. Please ensure that the function definition of `getDetail` is updated to handle `mySignedId` and `req?.userId`.

```diff
- } = getDetail(req, b, mySignedId);
+ } = getDetail(req, b, mySignedId, req?.userId);
```

File: controllers/feeds/getAnonymousFeedChat.js [hunk 2]
```diff
@@ -57,6 +58,7 @@ const getAnonymousFeedChatService = async (req, res) => {
           totalComment: totalComment + commentLevel2 + total3,
           isOwnPost,
           isOwnSignedPost,
+          isOwnAnonymousPost,
```
Comment (bastian-purrer) [inline controllers/feeds/getAnonymousFeedChat.js:61]: Image description   CodeRabbit

The addition of `isOwnAnonymousPost` in the returned object is a good change as it provides more context about the post ownership. However, please ensure that this variable is properly initialized and used where necessary.

```diff
+ isOwnAnonymousPost,
```

File: controllers/feeds/getFeedChat.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ const mappingFeed = (req, feeds) => {
   return mapping;
 };
 
-const getDetail = (req, b, id) => {
+const getDetail = (req, b, id, anonymousId) => {
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:42]: Image description   CodeRabbit

The function `getDetail` now has a new parameter `anonymousId`. This change seems to be consistent with the PR title and description. However, it's important to ensure that all calls to this function are updated to pass this new argument.
```diff
- const getDetail = (req, b, id) => {
+ const getDetail = (req, b, id, anonymousId) => {
```

File: controllers/feeds/getFeedChat.js [hunk 4]
```diff
@@ -51,7 +51,8 @@ const getDetail = (req, b, id) => {
   let actor = typeof b.object === 'object' ? b.object.actor : b.actor;
   const isAnonym = typeof b.object === 'object' ? b.object.anonimity : b.anonimity;
   const isOwnSignedPost = actor?.id === id;
-  const isOwnPost = actor?.id === req.userId || actor?.id === id;
+  const isOwnAnonymousPost = actor?.id === anonymousId;
+  const isOwnPost = isOwnAnonymousPost || isOwnSignedPost;
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:55]: Image description   CodeRabbit

The logic for determining if a post is owned by the current user has been updated. Previously, it was checking if the actor's id matches either `req.userId` or `id`. Now, it checks if the actor's id matches either `anonymousId` or `id`. This change seems to be in line with the PR's purpose of separating anonymous and signed comment reactions.
```diff
- const isOwnPost = actor?.id === req.userId || actor?.id === id;
+ const isOwnAnonymousPost = actor?.id === anonymousId;
+ const isOwnPost = isOwnAnonymousPost || isOwnSignedPost;
```

File: controllers/feeds/getFeedChat.js [hunk 5]
```diff
@@ -79,6 +80,7 @@ const getDetail = (req, b, id) => {
     isAnonym,
     isOwnSignedPost,
     isOwnPost,
+    isOwnAnonymousPost,
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:83]: Image description   CodeRabbit

The variable `isOwnAnonymousPost` is now included in the returned object. This seems to be consistent with the PR's purpose of distinguishing between anonymous and signed posts.
```diff
+ isOwnAnonymousPost,
```

File: controllers/feeds/getFeedChat.js [hunk 6]
```diff
@@ -120,7 +122,7 @@ const finalize = (req, id, myReaction, newGroup, activity_id, constantActor, kar
   if (myReaction) {
     myReaction = {
       ...myReaction,
-      isOwningReaction: req.userId === myReaction.user_id || myReaction.user_id === id
+      isOwningReaction: req.userId === myReaction.user_id
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:125]: Image description   CodeRabbit

The condition for `isOwningReaction` has been simplified to only check if `req.userId` equals `myReaction.user_id`. This change seems to be in line with the PR's purpose of separating anonymous and signed comment reactions.
```diff
- isOwningReaction: req.userId === myReaction.user_id || myReaction.user_id === id
+ isOwningReaction: req.userId === myReaction.user_id
```

File: controllers/feeds/getFeedChat.js [hunk 7]
```diff
@@ -234,10 +236,12 @@ const getFeedChatService = async (req, res) => {
         constantActor,
         isAnonym,
         isOwnPost,
+        isOwnSignedPost,
+        isOwnAnonymousPost,
         message,
         actor,
         showToast = false
-      } = getDetail(req, b, myAnonymousId.user_id);
+      } = getDetail(req, b, req?.userId, myAnonymousId?.user_id);
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:244]: Image description   CodeRabbit

The call to `getDetail` now includes two new arguments: `req?.userId` and `myAnonymousId?.user_id`. Also, two new variables `isOwnSignedPost` and `isOwnAnonymousPost` are now being destructured from the returned object. These changes seem to be consistent with the PR's purpose of distinguishing between anonymous and signed posts.
```diff
- } = getDetail(req, b, myAnonymousId.user_id);
+ } = getDetail(req, b, req?.userId, myAnonymousId?.user_id);
```

File: controllers/feeds/getFeedChat.js [hunk 8]
```diff
@@ -261,6 +265,8 @@ const getFeedChatService = async (req, res) => {
           isSeen: b.isSeen,
           totalComment: totalComment + commentLevel2 + total3,
           isOwnPost,
+          isOwnAnonymousPost,
+          isOwnSignedPost,
```
Comment (bastian-purrer) [inline controllers/feeds/getFeedChat.js:269]: Image description   CodeRabbit

The variables `isOwnAnonymousPost` and `isOwnSignedPost` are now included in the returned object. This seems to be consistent with the PR's purpose of distinguishing between anonymous and signed posts.
```diff
+ isOwnAnonymousPost,
+ isOwnSignedPost,
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 9638e663bf40e65529bb3f8373aa4dba98b5bf60 and 71b13d538d454255fc223ed8ccff1f15a427ed5d commits.
</details>

<details>
<summary>Files selected (2)</summary>

* controllers/feeds/getAnonymousFeedChat.js (2)
* controllers/feeds/getFeedChat.js (6)
</details>








<details>
<summary>Review comments generated (8)</summary>

* Review: 8
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by n-permana — APPROVED** [review]

**Review by brm-stnd — APPROVED** [review]

**Pull request merged**